src/web/client/common/authenticationProvider.ts

```diff
@@ -3,63 +3,121 @@
  * Licensed under the MIT License. See License.txt in the project root for license information.
  */
```

Comment:
I see you have touched lot of core files and I am pretty sure lot of UTs are affected. What's your plan on adding UTs for them?

Reply:
Can we make a practice to add UTs for all newly added methods other it will always in a catch-up game for writing UTs.

Reply:
Will take this up as follow up task